bip-0000.mediawiki

```diff
@@ -0,0 +1,210 @@
+
+* Let ''P<sub>0</sub> = HASH(outpoints_hash·b<sub>scan</sub>·A || 0)·G + B<sub>spend</sub>''
+* Compute ''P<sub>0m</sub> = P<sub>0</sub> + m·G'' for each ''m''
+* For each ''P<sub>0m</sub>'' in {''P<sub>00</sub> … P<sub>0m</sub>''}, check if any are present in the transaction outputs
+
+== Specification ==
+
+In what follows, we use the notation from [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#specification BIP340].
+
+=== Address encoding ===
+
+A silent payment address is constructed in the following manner:
+
+* Let ''HRP = sp''
+* Let ''v = a single version byte, set to 0''
+* Let ''B<sub>scan</sub>, b<sub>scan</sub> = Recipient’s [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki BIP340] scan key pair''
+* Let ''B<sub>spend</sub>, b<sub>spend</sub> = Recipient’s [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki BIP340] spend key pair''
+* Let ''B<sub>m</sub> = B<sub>spend</sub> + m·G'', where ''m'' an optional integer tweak for labeling
+** In the case of ''m'' = 0, no label is applied and ''B<sub>m</sub> = B<sub>spend</sub>''
+* The final address is a 66 byte address of the form ''HRP || v || B<sub>m</sub> || B<sub>scan</sub>''
+
+=== Outpoint hash ===
+
+The sender and receiver MUST calculate an outpoint hash for the transaction in the following manner:
+
+* Collect each ''outpoint'' from the transaction, where an outpoint is the 36 byte little-endian representation of the previous output transaction id and vout index
+* Sort the outpoints alphanumerically, in ascending order
+* Let ''outpoints = outpoint<sub>0</sub> || outpoint<sub>1</sub> || … || outpoint<sub>n</sub>''
+* Let ''outpoint_hash = SHA256(outpoints), truncated to the first 8 bytes''<ref name="why_8byte_hash">'''How was an 8 byte hash chosen?''' Reference the birthday problem and numbers there, etc (TODO)</ref>
+
+=== Sender ===
+
+==== Selecting inputs ====
+
+The sending wallet performs coin selection as usual with the added restriction that the sending wallet must be able to produce a signature for each UTXO. This excludes any UTXOs not requiring a signature for spending as these do not represent elliptic curve points. Given that these types of scripts are not widely used, no changes to coin selection will be required for the majority of wallets.<ref name="selecting_inputs">'''How to handle multisig and complex scripts''' TODO</ref>
+
+==== Creating outputs ====
+
+After the inputs have been selected, the sender creates the receiver's output(s) in the following manner:
+
+* Let ''a = a<sub>0</sub> + a<sub>1</sub> + … a<sub>i</sub>'', where each ''a<sub>i</sub>'' is the 256-bit private key corresponding to the ''ith'' input public key of the transaction
+* Compute the tweak:
+** Let ''t<sub>0</sub> = SHA256(SHA256(outpoint_hash·a·B<sub>scan</sub> || 0))''
+** Optionally, if more outputs are desired:
+*** Starting from ''n = 1''
+**** Let ''t<sub>n</sub> = SHA256(SHA256(outpoint_hash·a·B<sub>scan</sub> || n))''
```

Comment:
same question as above regarding `mod p`: should we multiply the scalars first, `outpoint_hash * a`, and do `mod p` before multiplying by `B`?